engine/src/main/java/org/terasology/engine/persistence/typeHandling/extensionTypes/ChunkMeshTypeHandler.java

```diff
@@ -44,7 +45,7 @@ protected PersistedData serializeNonNull(ChunkMesh value, PersistedDataSerialize
             directBuffer.rewind();
             asBuffers.add(directBuffer);
         }
-        ChunkMesh result = new ChunkMeshImpl();
+        MutableChunkMesh result = new ChunkMeshImpl();
         for (ChunkMesh.RenderType renderType : ChunkMesh.RenderType.values()) {
             result.getVertexElements(renderType).buffer.replace(asBuffers.remove(0));
```

Comment:
this probably should be changed. these are raw buffers for opengl so they follow the endiness of the machine if two machines have different endingess then it will be garbled. network order is little endian.